examples/test_util/EclFile.cpp

```diff
@@ -0,0 +1,495 @@
+        throw std::runtime_error(message);
+    }
+    
+    arrName=tmpStrName;
+    
+    if (tmpStrType=="INTE")
+        arrType=EIOD::INTE;
+    else if (tmpStrType=="REAL")    
+        arrType=EIOD::REAL;
+    else if (tmpStrType=="DOUB")    
+        arrType=EIOD::DOUB;
+    else if (tmpStrType=="CHAR")    
+        arrType=EIOD::CHAR;
+    else if (tmpStrType=="LOGI")    
+        arrType=EIOD::LOGI;
+    else
+        throw std::runtime_error("Error, unknown array type '" + tmpStrType +"'");
+    
+}
+
+
+unsigned long int sizeOnDisk(const int num, const EIOD::eclArrType &arrType){
+    
+    unsigned long int size=0;
+
+    if (arrType==EIOD::MESS){
+        if (num>0) {
+            std::string message="In routine calcSizeOfArray, type MESS can not have size > 0";
+            throw std::invalid_argument(message);
+        }
+    } else {
+
+        auto sizeData = block_size_data(arrType);
+
+	int sizeOfElement = std::get<0>(sizeData);
+        int maxBlockSize = std::get<1>(sizeData);
+        int maxNumberOfElements=maxBlockSize/sizeOfElement;
+       
+        size=num*sizeOfElement;
+        size=size + ((num-1) / maxNumberOfElements)*2*EIOD::sizeOfInte;    // 8 byte (two integers) every 1000 element        
+        
+        if (num>0){
+            size=size+2*EIOD::sizeOfInte;
+        }
+    }
+         
+    return size;
+}
+    
+
+
+
+std::vector<int> readBinaryInteArray(std::fstream &fileH,const int size){
+
+    std::vector<int> arr;
+    int value;
+    int num,dhead,dtail,rest;
+    
+    auto sizeData = block_size_data(EIOD::INTE);
+
+    int sizeOfElement = std::get<0>(sizeData);
+    int maxBlockSize = std::get<1>(sizeData);
+    int maxNumberOfElements=maxBlockSize/sizeOfElement;
+
+    rest=size;
+    
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+   
+    while (rest > 0) {   
+ 
+        fileH.read((char *)&dhead, sizeof(dhead));
+        dhead=flipEndianInt(dhead);
+        
+        num=dhead/sizeOfElement;
+
+        if ((num > maxNumberOfElements) || (num < 0)){
+            std::string message="Error reading binary inte data, inconsistent header data or incorrect number of elements";
+            throw std::runtime_error(message);
+        }
+       
+        for (int i=0;i<num;i++){
+            fileH.read((char *)&value, sizeOfElement);
+            value=flipEndianInt(value);
+            arr.push_back(value);
+        }
+       
+        rest=rest-num;
+       
+        if (((num < maxNumberOfElements) && (rest!=0)) || ((num==maxNumberOfElements) && (rest < 0))) {
+            std::string message="Error reading binary integer data, incorrect number of elements";
+            throw std::runtime_error(message);
+        }
+       
+        fileH.read((char *)&dtail, sizeof(dtail));
```

Comment:
Use of `sizeof( )` here hints at a flexibility which is not for real - the file format dictates that you should always read a for byte integer here; i.e. changing to `std::size_t dtail;` would create havoc.

Reply:
Please let me know what you would like me to use instead of sizeof(dtail)

Reply:
This is no big deal; my only point is that the use of `sizeof( )` communicates a flexibility which is not actually real - the file format dictates that this should be a 32 bit integer - in the `readBinaryHeader()` function you have hardcoded these numbers to `4` and `8`; I would maybe use a variable:
```
int sizeof_int = 4;
```
anyway - no big deal.